core/src/main/java/fr/sncf/osrd/conflicts/Conflicts.kt

```diff
@@ -239,3 +235,47 @@ private fun <ReqT : ResourceRequirement> detectRequirementConflicts(
     }
     return conflictGroups
 }
+
+fun mergeConflicts(conflicts: List<Conflict>): List<Conflict>{
+
```

Comment:
This is overly complex, both in terms of readability and algorithmic time-complexity.

This may have been unclear in the issue, but we really want to group conflicts by sets of conflicting trains, using a HashMap of `[TrainA, TrainB, etc]` (keep the trains sorted in these, or better, use a set if that's possible in kotlin?)

(You want to keep the conflicts separated by types too, so maybe have two hashmaps ;) )

Then for each hashmap entry, sort the resulting list of conflicts, and then it's a simple matter of iterating through it in a single pass.

You can then have a second pass where you remove spacing conflict that overlap a routing conflict.